### DISCUSSION TITLE
[Adonis 4.1] Event.emit() not working inside a class constuctor

### DISCUSSION THREAD

**Discussion by wemersonrv** [adonisjs/core#2306] [Help]
Hello guys... i need to emit an event from a class constructor but not works... does not gives me any error, but not works... and when i call anoter event **after** class constructor runs, it works fine...

There is a way to make Event.emit() works from a class constructor?

**app/services/MyCustomClass.js**
```JS
'use strict'

const Event = use('Event')

class MyCustomClass { 
   constructor() {

      // Nothing happens here
      Event.emit( 'some::event',  {
         some: 'data',
      })

   }

   someMethod(){

      // Event is emitted, and it works fine...
      Event.emit( 'some::event',  {
         some: 'data',
      })

   }
   
}

module.exports = MyCustomClass
```

**start/events.js**
```js
const Event = use('Event')

Event.on('some::event', (data) => {
  console.log('some::event event fired...', data)
})

```

**Comment by McSneaky** [comment]
Where and how are you calling your class constructor (initializing new class instance)?

**Reply by wemersonrv** [reply]
Hello,

It's rgistered as a `ioc.singleton` in a Service Provider.
